### PR TITLE
Added layersupport to waypoints!!

### PR DESCRIPTION
**Pull request by daanbreur**
It will now save what layer its made on!

This PR is related to: https://trello.com/c/CyY9GyV9

If you make a waypoint on the `wires` layer and click the waypoint button it will teleport you to the location and set the layer to the `wires` layer. And ofcourse Vise versa 

I am kinda bad in explaining.

### CODE REVIEW

File: src/js/game/hud/parts/waypoints.js [hunk 1]
```diff
@@ -228,6 +231,7 @@ export class HUDWaypoints extends BaseHUDPart {
      * @param {Waypoint} waypoint
      */
     moveToWaypoint(waypoint) {
+        this.root.currentLayer = waypoint.layer;
```
Comment (tobspr) [inline src/js/game/hud/parts/waypoints.js:234]: You'll have to migrate old savegames, otherwise it will break here since layer is null then

**Review by tobspr — CHANGES_REQUESTED** [review]

File: src/js/game/hud/parts/waypoints.js (hunk 1 above)
Comment (daanbreur) [reply src/js/game/hud/parts/waypoints.js:234]: oh oops i completely forgot! I'll fix it!

**Comment by daanbreur** [comment]
Added Migrator for v1006 to v1007!
This should work!

File: src/js/savegame/schemas/1007.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { createLogger } from "../../core/logging.js";
+import { SavegameInterface_V1006 } from "./1006.js";
+
+const schema = require("./1007.json");
+const logger = createLogger("savegame_interface/1006");
```
Comment (tobspr) [inline src/js/savegame/schemas/1007.js:5]: Should be 1007 :)

File: src/js/savegame/schemas/1007.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    static migrate1006to1007(data) {
+        logger.log("Migrating 1006 to 1007");
+        const dump = data.dump;
+        if (!dump) {
+            return true;
+        }
+
+        const waypoints = dump.waypoints.waypoints;
+
+        // clear ejector slots
```
Comment (tobspr) [inline src/js/savegame/schemas/1007.js:28]: I think you forgot to remove this comment :)

File: src/js/game/hud/parts/waypoints.js [hunk 4]
```diff
@@ -98,6 +100,7 @@ export class HUDWaypoints extends BaseHUDPart {
                 label: null,
                 center: { x: 0, y: 0 },
                 zoomLevel: 3,
+                layer: layers[0],
```
Comment (tobspr) [inline src/js/game/hud/parts/waypoints.js:103]: You can hardcode this to "regular" or you use `gBuildingRegistry.findByClass(MetaHubBuilding).getLayer()`

**Review by tobspr — CHANGES_REQUESTED** [review]

File: src/js/game/hud/parts/waypoints.js (hunk 4 above)
Comment (daanbreur) [reply src/js/game/hud/parts/waypoints.js:103]: I did hardcode it to "regular" but vscode and gulp where screaming at me... that its not the Layer type

**Comment by tobspr** [comment]
Looks good! Since this deals with savegames, I want to make sure its tested well so it might take a bit, but thanks for the work already! :)

**Comment by daanbreur** [comment]
Okey! Thats reasonable xD.

**Comment by daanbreur** [comment]
Sorry for being inpatient. But can i get an update on when you can merge it?

**Comment by tobspr** [comment]
probably not before the wires update, depends on how much time I have left! :)

**Comment by daanbreur** [comment]
Okey!

**Comment by daanbreur** [comment]
Sooooooooon™️ 19 days ago..

**Comment by tobspr** [comment]
It's on my todo list :P

**Comment by daanbreur** [comment]
Yey

**Comment by daanbreur** [comment]
Boys we did it! Its merged. Thanks

**Pull request merged**